File: src/games/stendhal/client/events/ItemUpgradeEvent.java
/***************************************************************************
 *                   (C) Copyright 2026 - PolanieOnLine                    *
 ***************************************************************************/
package games.stendhal.client.events;

import games.stendhal.client.entity.Entity;
import games.stendhal.client.gui.ItemUpgradeWindow;

/** Opens or refreshes the desktop item-upgrade window. */
final class ItemUpgradeEvent extends Event<Entity> {
	@Override
	public void execute() {
		ItemUpgradeWindow.show(event);
	}
}

File: src/js/stendhal/event/ItemUpgradeEvent.ts
/***************************************************************************
 *                   (C) Copyright 2026 - PolanieOnLine                    *
 ***************************************************************************/

import { RPEvent } from "marauroa";
import { ItemUpgradeDialog } from "../ui/dialog/ItemUpgradeDialog";

/** Opens or refreshes the web Item Upgrades 2.0 dialog. */
export class ItemUpgradeEvent extends RPEvent {
	public execute(_entity: any): void {
		ItemUpgradeDialog.show(this as any);
	}
}
